### DISCUSSION TITLE
classes/class_httprequest

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#3] [User-contributed notes]
# classes/class_httprequest

Inherits: Node< Object A node with the ability to send HTTP(S) requests. Description: A node with the ability to send HTTP requests. Uses HTTPClient internally. Can be used to make HTTP requests, i...

https://docs.godotengine.org/en/latest/classes/class_httprequest.html?highlight=HTTPRequest

**Comment by Swarkin** [comment]
I made an `AwaitableHTTPRequest` Node to make HTTP requests more convenient and just in one line:
https://github.com/Swarkin/Godot-AwaitableHTTPRequest

**Comment by Swarkin** [comment]
Its worth mentioning that Godot **4.4** will **remove** support for the old TLS 1.0 and 1.1 protocols according to the documentation of HTTPClient.

**Reply by Swarkin** [reply]
In turn it adds support for TLS 1.3

**Reply by Calinou** [reply]
Where did you see that TLS 1.0/1.1 support were removed?

Note that these protocols are known to be unsafe and many other clients already forbid connecting to them by default.

**Reply by Swarkin** [reply]
https://docs.godotengine.org/en/latest/classes/class_httpclient.html
> "Note: TLS support is currently limited to TLSv1.2 and TLSv1.3."

**Comment by bruno-pmakers** [comment]
Should the http be queued free at some point after the request is made?

**Reply by bruno-pmakers** [reply]
Something like

```gdscript
func _ready():
	# Create an HTTP request node and connect its completion signal.
	var http_request = HTTPRequest.new()
	add_child(http_request)
	http_request.request_completed.connect(self._http_request_completed)

	# Perform a GET request. The URL below returns JSON as of writing.
	var error = http_request.request("https://httpbin.org/get")
	if error != OK:
		push_error("An error occurred in the HTTP request.")

	# Perform a POST request. The URL below returns JSON as of writing.
	# Note: Don't make simultaneous requests using a single HTTPRequest node.
	# The snippet below is provided for reference only.
	var body = JSON.new().stringify({"name": "Godette"})
	error = http_request.request("https://httpbin.org/post", [], HTTPClient.METHOD_POST, body)
	if error != OK:
		push_error("An error occurred in the HTTP request.")

    http_request.queue_free() ## <<-------- Add this
```

**Reply by Calinou** [reply]
You can free it if you don't need to make more requests later, but it's not strictly needed.
